Composer/packages/client/src/pages/design/DebugPanel/TabExtensions/WatchTab/WatchTabContent.tsx

```diff
@@ -0,0 +1,326 @@
+      const abortController = new AbortController();
+      (async () => {
+        try {
+          const variables = await getMemoryVariables(currentProjectId, { signal: abortController.signal });
+          setMemoryVariablesPayload({ kind: 'property', data: { properties: variables } });
+        } catch (e) {
+          // error can be due to abort
+        }
+      })();
+    }
+  }, [currentProjectId]);
+
+  const mostRecentBotState = useMemo(() => {
+    const botStateTraffic = rawWebChatTraffic.filter(
+      (t) => t.trafficType === 'activity' && t.activity.type === 'trace' && t.activity.name === 'BotState'
+    ) as ConversationActivityTrafficItem[];
+    if (botStateTraffic.length) {
+      return botStateTraffic[botStateTraffic.length - 1];
+    }
+  }, [rawWebChatTraffic]);
+
+  const onSelectPath = useCallback(
+    (variableId: string, path: string) => {
+      const watchedVar = Object.values(watchedVars).find((varPath) => varPath === path);
+      if (watchedVar) {
+        // the variable is already being watched, so display a validation error under the picker
+        setPickerErrorMessages({
+          ...pickerErrorMessages,
+          [variableId]: formatMessage('You are already watching this property.'),
+        });
+      } else {
+        setWatchedVars({
+          ...watchedVars,
+          [variableId]: path,
+        });
+        // clear any error messages for the variable
+        delete pickerErrorMessages[variableId];
+        setPickerErrorMessages({ ...pickerErrorMessages });
+      }
+    },
+    [pickerErrorMessages, watchedVars]
+  );
+
+  // we need to refresh the details list every time a new bot state comes in
+  const refreshedWatchedVars = useMemo(() => {
+    return Object.entries(watchedVars).map(([key, value]) => {
+      return {
+        key,
+        value,
+      };
+    });
+  }, [mostRecentBotState, pickerErrorMessages, watchedVars]);
+
+  const renderRow = useCallback((props?: IDetailsRowProps) => {
+    if (props) {
+      return <DetailsRow {...props} styles={rowStyles} />;
+    }
+    return null;
+  }, []);
+
+  const renderColumn = useCallback(
+    (item: { key: string; value: string }, index: number | undefined, column: IColumn | undefined) => {
+      if (column && index !== undefined) {
+        if (column.key === NameColumnKey) {
+          // render picker
+          return (
+            <WatchVariablePicker
+              key={item.key}
+              errorMessage={pickerErrorMessages[item.key]}
+              path={item.value}
+              payload={memoryVariablesPayload}
+              variableId={item.key}
+              onSelectPath={onSelectPath}
+            />
+          );
+        } else if (column.key === ValueColumnKey) {
+          // render the value display
+          if (mostRecentBotState) {
+            const value = getValueFromBotTraceScope(item.value, mostRecentBotState?.activity);
+            if (value !== null && typeof value === 'object') {
```

Comment:
how are we rendering the value `null`

Reply:
@sorgh and I chatted about this offline and here is an example of a table that is watching variables that return an object, "undefined", "null", or are not available:
![image](https://user-images.githubusercontent.com/39317542/122978338-42b38000-d34b-11eb-8ac9-8c3bba30b7c5.png)

"unavailable" in Segoe gray 130 (#605E5C); "undefined" and "null"  appear the same as the other values (same monospace font families and color coding).

Will this work?

Reply:
That should work for us. Also, unrelated, boolean would just be rendered as true or false without the string interpolation "true".